lightkube/core/dataclasses_dict.py

```diff
@@ -96,6 +96,13 @@ class DataclassDictMixIn:
     _prop_to_json: typing.Dict = None
     _valid_params: typing.Set = None
 
+    def __setattr__(self, name, value):
+        if not name.startswith('_'):
+            _lazy_values = getattr(self, "_lazy_values", {})
+            if name in _lazy_values and not isinstance(value, LazyAttribute):
+                del self._lazy_values[name]
+        self.__dict__[name] = value
+
```

Comment:
@gtsystem I believe this will address #44 but i'd like your feedback on if you'd like this resolved this way